core/src/main/scala/io/projectglow/bgen/BgenRowToInternalRowConverter.scala

```diff
@@ -71,21 +74,40 @@ class BgenRowToInternalRowConverter(schema: StructType) extends GlowLogging {
     new RowConverter[BgenRow](schema, fns.toArray)
   }
 
-  private def makeGenotypeConverter(gSchema: StructType): RowConverter[BgenGenotype] = {
+  private def makeGenotypeConverter(
+      gSchema: StructType,
+      hardCallsThreshold: Option[Double]): RowConverter[(Int, BgenGenotype)] = {
     val functions = gSchema.map { field =>
-      val fn: RowConverter.Updater[BgenGenotype] = field match {
+      val fn: RowConverter.Updater[(Int, BgenGenotype)] = field match {
         case f if structFieldsEqualExceptNullability(f, sampleIdField) =>
           (g, r, i) => {
```

Comment:
Can you expand the signature so that it looks like `((numAlleles, g), r, i)`? The `._1` and `._2` are hard to read.